weblate/middleware.py

```diff
@@ -242,6 +242,11 @@ def __call__(self, request):
         if "://" in settings.SOCIAL_AUTH_AUTH0_IMAGE:
             domain = urlparse(settings.SOCIAL_AUTH_AUTH0_IMAGE).hostname
             image.add(domain)
+        
+        # Cloudflare use for security
+        if "://" in settings.CLOUDFLARE_URL:
```

Comment:
This will crash if the setting doesn't exist.

Reply:
This config for some domain which is use cloudflare. it prevent cloudflare's script was block by the weblate CSP limit (while DEBUG=0).
Does it make a define in `weblate/settings_docker.py` in `Third party services integration` can solve the crash problem?

Reply:
The setting should either not be required (check its presence) or have default value (preferred). We use https://django-appconf.readthedocs.io/en/latest/ to provide default values for settings.

Also, I can see other issues:

* It doesn't seem to be Cloudflare specific, so I see no good reason for using CLOUDFLARE in the name.
* The URL is not really used as URL, all you use is a domain name. Why not have just this one in the setting?

How about having `CSP_SCRIPT_SRC` configuration, which would allow list of strings and those would be added to the CSP?

Reply:
I'm agree with that, `CSP_SCRIPT_SRC` will be better.